### PR TITLE
Refactor how we set/restore ownership of drupal settings file to allow reuse...

### PR DESCRIPTION
**Pull request by jasonhildebrand**
...in containers based on this one.

The net effect of this PR on its own is none - everything works as before, I have simply factored out these two functions.

I'll be submitting a PR against isle-dc shortly where I make use of these functions to fix an issue occurring there.

### CODE REVIEW

**Comment by jasonhildebrand** [comment]
Here is the issue this refactoring helps to fix: https://github.com/Islandora-Devops/isle-dc/pull/208

File: drupal/rootfs/etc/islandora/utilities.sh [hunk 1]
```diff
@@ -336,9 +369,7 @@ function update_settings_php {
     fi
 
     # Restore owner/group to previous value
-    if [ ! -z "${previous_owner_group}" ]; then
-        chown "${previous_owner_group}" "${site_directory}/settings.php"
-    fi
+    restore_settings_ownership ${site} ${previous_owner_group}
 
     # Restrict access to settings.php
     chmod 444 "${site_directory}/settings.php"
```
Comment (nigelgbanks) [inline drupal/rootfs/etc/islandora/utilities.sh:375]: We should move the correction of permissions into `restore_settings_ownership`
```bash
# Restrict access to settings.php
chmod 444 "${site_directory}/settings.php"
```

File: drupal/rootfs/etc/islandora/utilities.sh [hunk 2]
```diff
@@ -284,6 +284,43 @@ function fedora_url {
...
+function allow_settings_modifications {
+    local site="${1}"; shift
+    local drupal_root=$(drush drupal:directory)
+    local subdir=$(drupal_site_env "${site}" "SUBDIR")
+    local site_directory=$(realpath "${drupal_root}/sites/${subdir}")
+
+    # send debug output to stderr because the caller typically captures output from this function.
+    #>&2 echo "adjusting ownership of ${site_directory}/settings.php"  
+    if [ -f "${site_directory}/settings.php" ]; then
+        previous_owner_group=$(stat -c "%u:%g" "${site_directory}/settings.php")
```
Comment (nigelgbanks) [inline drupal/rootfs/etc/islandora/utilities.sh:298]: On review I don't know why we'd ever have an owner/group other than nginx:nginx (100:101)? With the possible exception if it's bind mounted. Possibly all files in `/var/www/drupal` have the hosts users group to allow for host edits or something... That being said it doesn't hurt anything. Just a thought as to why did we do this in the first place.

File: drupal/rootfs/etc/islandora/utilities.sh [hunk 3]
```diff
@@ -284,6 +284,43 @@ function fedora_url {
...
+    local site="${1}"; shift
+    local previous_owner_group="${1}"; shift
+    local drupal_root=$(drush drupal:directory)
+    local subdir=$(drupal_site_env "${site}" "SUBDIR")
+    local site_directory=$(realpath "${drupal_root}/sites/${subdir}")
+
+    # Restore owner/group to previous value
+    if [ ! -z "${previous_owner_group}" ]; then
+        #echo "restoring ${site_directory}/settings.php"
+        #echo "previous_owner_group ${previous_owner_group}"
```
Comment (nigelgbanks) [inline drupal/rootfs/etc/islandora/utilities.sh:318]: We can remove the commented out `#echo` statements.

**Review by nigelgbanks — CHANGES_REQUESTED** [review]
Good stuff, just a small gotcha with the restoration of perms being outside of the restore function.

File: drupal/rootfs/etc/islandora/utilities.sh (hunk 2 above)
Comment (jasonhildebrand) [reply drupal/rootfs/etc/islandora/utilities.sh:298]: Good catch re: permissions.  I've moved this into the function.

Restoring ownership is only important when it's bind mounted.  It ensures the file remains readable/writable by the host user.  I've added a comment in the code to this effect.

File: drupal/rootfs/etc/islandora/utilities.sh (hunk 3 above)
Comment (jasonhildebrand) [reply drupal/rootfs/etc/islandora/utilities.sh]: Agreed - I've taken them out.

File: drupal/rootfs/etc/islandora/utilities.sh (hunk 1 above)
Comment (jasonhildebrand) [reply drupal/rootfs/etc/islandora/utilities.sh]: Yes - good catch. I've made this change.

**Review by nigelgbanks — APPROVED** [review]

**Pull request merged**